document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -257,12 +258,31 @@ protected String parseNonCompositeFilter(Filter filter, Params.Builder paramsBui
         // use two question marks (??) as escape sequence.
         // This is only true for postgres jdbc driver.
         // https://jdbc.postgresql.org/documentation/head/statement.html
-        filterString.append(" ?? ");
+        sqlOperator = " ?? ";
+        if (OUTER_COLUMNS.contains(fieldName)) {
+          // Operator is only supported on jsonb datatype
+          // i.e. fields inside `document` column
+          throw new UnsupportedOperationException("Operation is not supported on this column.");
```

Comment:
What should be the behaviour for OUTER_COLUMNS in postgres schema,
should it throw exception or just ignore this filter since outer columns will always be defined in all records, as per current definition.